### DISCUSSION TITLE
Using development Snapshots

### DISCUSSION THREAD

**Discussion by devmrsmith** [cashapp/paparazzi#898] [General]
Hey

I'd like to try and use the development Snapshots which have the most recent (post 1.2.0) bugfixes in them as detailed here:

https://cashapp.github.io/paparazzi/#releases

The only thing is I don't know what changes I need to make in gradle to replace the standard dependency and plugin declarations.

Can anybody help?

Many thanks!

**Comment by TWiStErRob** [comment]
Just replace the version numbers (`1.2.0`) to the snapshot: `1.3.0-SNAPSHOT` and make sure you added the snapshot repository to all `repositories` blocks (Paparazzi is both Gradle plugin and runtime dependency). You can check the latest version by looking at the directory listing: https://oss.sonatype.org/content/repositories/snapshots/app/cash/paparazzi/paparazzi/

**Reply by devmrsmith** [reply]
Perfect ... thanks Robert ... much appreciated!

**Comment by devmrsmith** [comment]
Hey @TWiStErRob - we've not anywhere with the development Snapshots as we can't upgrade our project to AGP 8. 

I'm going to try and build the Paparazzi project (adjusting AGP down to 7.3.0) and then import it ... only thing is I've never done it for a plugin before. Any pointers?

Cheers!

**Reply by TWiStErRob** [reply]
That may or may not work... your best bet is to try. The changes should be trivial, find the numbers you want to change and then try to build.

There are changes made that you might need to revert, because the assumption on `master` is Gradle 8 / AGP 8 now. See for example the last commit: https://github.com/cashapp/paparazzi/commit/2841f89aacd3160433937c0ebdb1506e69d68eea, there might be others too, the [list is long](https://github.com/cashapp/paparazzi/compare/1.2.0...master). _Depending on what bugfixes you need_, you might have better luck starting from the `1.2.0` tag and finding a commit that has what you need / what doesn't break your project, and maybe cherry-pick things on top. Beware that PR squashing wasn't used consistently until recently.
